python/lsst/daf/butler/script/exportCalibs.py

```diff
@@ -0,0 +1,147 @@
+    """
+    butler = Butler(repo, writeable=False)
+
+    calibTypes = ["bias", "dark", "flat", "defects", "linearity", "bfk", "ptc", "crosstalk"]
+
+    collectionsToExport = []
+    datasetsToExport = []
+
+    for collection in collections:
+        log.info("Checking collection: %s", collection)
+
+        # Get collection information.
+        collectionsToExport.append(collection)
+        collectionType = butler.registry.getCollectionType(collection)
+        if collectionType == CollectionType.CHAINED:
+            # Iterate over the chain:
+            collectionsToExport.append(collection)
+            collectionRecords = butler.registry.getCollectionChain(collection)
+
+            for child in collectionRecords:
+                childType = butler.registry.getCollectionType(child)
+                collectionsToExport.append(child)
+                if childType == CollectionType.CALIBRATION:
+                    exportCollections, exportDatasets = parseCalibrationCollection(butler.registry,
+                                                                                   child,
+                                                                                   calibTypes)
+                    collectionsToExport.extend(exportCollections)
+                    datasetsToExport.extend(exportDatasets)
+        elif collectionType == CollectionType.CALIBRATION:
+            exportCollections, exportDatasets = parseCalibrationCollection(butler.registry,
+                                                                           collection,
+                                                                           calibTypes)
+            collectionsToExport.append(collection)
+            collectionsToExport.extend(exportCollections)
+            datasetsToExport.extend(exportDatasets)
+
+    if os.path.exists(directory):
+        raise RuntimeError(f"Export directory exists: {directory}")
+    os.makedirs(directory)
+    with butler.export(directory=directory, format="yaml", transfer="auto") as export:
+        collectionsToExport = list(set(collectionsToExport))
+        datasetsToExport = list(set(datasetsToExport))
+
+        for exportable in collectionsToExport:
+            try:
+                export.saveCollection(exportable)
+            except Exception as e:
+                log.warning("Did not save collection %s due to %s.", exportable, e)
+
+        try:
+            log.info("Saving dataset(s)")
+            export.saveDatasets(datasetsToExport)
+        except Exception as e:
+            log.warning("Did not save datasets %s due to %s.", exportable, e)
+
+    sortedDatasets = sorted(datasetsToExport, key=lambda x: x.datasetType.name)
+
+    return Table({"calibrationType": [ref.datasetType.name for ref in sortedDatasets],
+                  "run": [ref.run for ref in sortedDatasets],
+                  "instrument": [ref.dataId["instrument"] for ref in sortedDatasets],
```

Comment:
Hard-coding this seems wrong to me. What if a calibration dataset type has an extra dimension? The robust answer seems to be that for all the dataset types involved -- work out all the relevant dimensions (which is not difficult to do by throwing all the dimensions of each datasetType into a set) and then for each dimension build the column and then add the column to this table.